rosetta/services/block.go

```diff
@@ -43,82 +60,100 @@ func (s *BlockAPIService) Block(
 
 	var (
 		requestedHeight int64
+		requestedTipSetKey filTypes.TipSetKey
 		tipSet *filTypes.TipSet
+		parentTipSet *filTypes.TipSet
 		err error
 	)
 
-	if request.BlockIdentifier != nil {
-		if request.BlockIdentifier.Index != nil {
-			requestedHeight = *request.BlockIdentifier.Index
-		} else if request.BlockIdentifier.Hash != nil {
-			decodedHeight, decodedTipSet, decErr := DecodeTipSetHash(*request.BlockIdentifier.Hash)
-			if decErr != nil {
-				return nil, decErr
+	if request.BlockIdentifier.Hash != nil {
+		decodedTipSet, decErr := DecodeTipSetHash(*request.BlockIdentifier.Hash)
```

Comment:
maybe a better name?
```suggestion
		decodedBlockCIDs, decErr := DecodeTipSetHash(*request.BlockIdentifier.Hash)
```

Reply:
Actually, `DecodeTipSetHash` should return a `TipSetKey` type, so no need for an extra after to get CIDs -> TipSetKey and name gets fixed also :)